### PR TITLE
FIX: simplify booksignal struct

### PR DESCRIPTION
**Pull request by c9s**

### CODE REVIEW

File: pkg/types/orderbook.go [hunk 1]
```diff
@@ -122,16 +122,16 @@ func (b *MutexOrderBook) Update(update SliceOrderBook) {
...
 const (
-	BookSignalSnapshot BookSignalType = "snapshot"
-	BookSignalUpdate   BookSignalType = "update"
+	BookSignalSnapshot BookSignalType = 1
+	BookSignalUpdate   BookSignalType = 2
 )
 
 type BookSignal struct {
 	Type BookSignalType
-	Book SliceOrderBook
```
Comment (c9s) [inline pkg/types/orderbook.go:134]: take off this big struct, could be too big to copy

File: pkg/types/rbtree.go [hunk 2]
```diff
@@ -437,33 +437,24 @@ func (tree *RBTree) PostorderOf(current *RBNode, cb func(n *RBNode) bool) {
 }
 
 func (tree *RBTree) CopyInorderReverse(limit int) *RBTree {
-	cnt := 0
 	newTree := NewRBTree()
-	tree.InorderReverse(func(n *RBNode) bool {
-		if cnt >= limit {
```
Comment (c9s) [inline pkg/types/rbtree.go:443]: missing limit > 0 check

File: pkg/types/rbtree.go [hunk 3]
```diff
@@ -473,3 +464,23 @@ func (tree *RBTree) Print() {
 		return true
 	})
 }
+
+func copyNodeFast(newTree *RBTree) func(n *RBNode) bool {
```
Comment (c9s) [inline pkg/types/rbtree.go:468]: make this functional

File: pkg/types/orderbook.go [hunk 4]
```diff
@@ -162,9 +162,7 @@ func (sb *StreamOrderBook) BindStream(stream Stream) {
 
 		sb.Load(book)
 		sb.EmitSnapshot(book)
-
-		// when it's snapshot, it's very important to push the snapshot signal to the caller
-		sb.C <- BookSignal{Type: BookSignalSnapshot, Book: book}
```
Comment (c9s) [inline pkg/types/orderbook.go:167]: remove the blocking behavior

**Review by bailantaotao — APPROVED** [review]

File: pkg/exchange/binance/parse.go [hunk 5]
```diff
@@ -510,6 +510,8 @@ func parseDepthEvent(val *fastjson.Value) (*DepthEvent, error) {
 		Symbol:        string(val.GetStringBytes("s")),
 		FirstUpdateID: val.GetInt64("U"),
 		FinalUpdateID: val.GetInt64("u"),
+		Bids: make(types.PriceVolumeSlice, 0, 50),
```
[the remark this answers is not in the document]
Comment (c9s) [reply pkg/exchange/binance/parse.go]: it's just a guess LOL

**Review by gx578007 — APPROVED** [review]

**Pull request merged**